packages/shared-components/cypress/e2e/user-menu/user-menu.cy.js

```diff
@@ -0,0 +1,55 @@
+import {UserMenuSteps} from "../../steps/user-menu/user-menu-steps";
+
+describe('User Menu', () => {
+  it('should render user menu', () => {
+    // Given I've opened a page with the user menu
+    UserMenuSteps.visit();
+
+    // And the user menu is visible
+    UserMenuSteps.getUserMenu()
+      .should('be.visible')
+      .and('have.text', 'john.doe');
+
+    // When, I click on the menu
+    UserMenuSteps.getUserMenu().click();
+
+    const dropdownOptions = ['My settings', 'Logout'];
+    // Then the dropdown should be displayed and have My Settings and Logout options
+    UserMenuSteps.getDropdown()
+      .children()
+      .each(($child, index) => {
+        expect($child.text()).to.equal(dropdownOptions[index]);
+      });
+
+    // When, I am logged in as external user
+    UserMenuSteps.setExternalUser();
+
+    // Then I expect to not see a logout option in the dropdown
+    UserMenuSteps.getDropdown().children()
+      .should('have.length', 1)
+      .and('not.contain', 'Logout');
+  });
+
+  it('should redirect to the correct URL when clicking on a dropdown option', () => {
+    // Given I've opened a page with the user menu
+    UserMenuSteps.visit();
+
+    // And the user menu is visible
+    UserMenuSteps.getUserMenu().should('be.visible');
+
+    // When, I click on the menu
+    UserMenuSteps.getUserMenu().click();
```

Comment:
It's better to abstract DOM access in the steps. This is their purpose. Instead of doing clicks, getting `dropdown.children().first().click()` etc., created methods like `openUserMenu()`, `selectSettings()`, etc.